packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ip-ip-agoraOrder.controller.js

```diff
@@ -20,14 +30,31 @@ export default class AgoraIpOrderCtrl {
     this.ALERT_ID = ALERT_ID;
     this.IP_TYPE = IP_TYPE;
     this.IP_TYPE_TITLE = IP_TYPE_TITLE;
+    this.ALLOWED_LANGUAGES = ALLOWED_LANGUAGES;
+    this.BASE_URL_SURVEY = BASE_URL_SURVEY;
     this.ipCatalog = ipCatalog;
+    this.coreConfig = coreConfig;
   }
 
   $onInit() {
     this.loading = {};
     this.user = this.$state.params.user;
     this.ipType = this.$state.params.ipType || this.ipType;
     this.getlowestPrice();
+    // Get default language
+    const defaultLanguage = Object.keys(this.ALLOWED_LANGUAGES).find(
+      (key) => this.ALLOWED_LANGUAGES[key].isDefault,
+    );
+    const userLanguage = this.coreConfig.getUserLanguage();
+    const languageToUse =
+      typeof this.ALLOWED_LANGUAGES[userLanguage] === 'object'
+        ? userLanguage
+        : defaultLanguage;
+    // Get user
+    const user = this.coreConfig.getUser();
+
+    // Build url for survey link
+    this.surveyUrl = `${this.BASE_URL_SURVEY}${languageToUse}&nic=${user.nichandle}`;
```

Comment:
I find the logic quite complicated.
Regarding the previous comment for the constant property, you can ease it in that way : 
```suggestion
    const userLanguage = this.coreConfig.getUserLanguage();
    const languageToUse = this.SURVEY_LANGUAGES.ALLOWED.includes(userLanguage) ? userLanguage : this.SURVEY_LANGUAGES.DEFAULT;
      
    // Get user
    const user = this.coreConfig.getUser();

    // Build url for survey link
    this.surveyUrl = `${this.BASE_URL_SURVEY}${languageToUse}&nic=${user.nichandle}`;
```
Maybe you can move it in a dedicated function as well